src/zrle.c

```diff
@@ -135,7 +135,7 @@ static void zrle_encode_packed_tile(struct vec* dst,
 	int bytes_per_cpixel = calc_bytes_per_cpixel(dst_fmt);
 
 	uint8_t cpalette[16 * 3];
-	pixel32_to_cpixel((uint8_t*)cpalette, dst_fmt, palette, src_fmt,
+	pixel_to_cpixel((uint8_t*)cpalette, dst_fmt, (uint8_t*)palette, src_fmt,
```

Comment:
The type of `palette` should be changed rather than typecasting it here

Reply:
I notice in copy tile 4 BPP is assumed, I think some modifications are necessary?

Reply:
Also notice that the dst buffer size is calculated from the source but the output format could be different.

Reply:
Anything that assumes a static pixel size must be changed.

The destination buffer reserved size should be calculated based on the total number of pixels and the destination pixel size. It doesn't matter if the buffer is oversized, it just shouldn't be undersized. We only send what's written to the buffer.

Reply:
ok, this makes sense, I will base the size of the destination format.

Reply:
This thread is still unresolved.

Reply:
Once I started making this change it cascaded to something much larger and I think Id like to fix this file in a separate PR.